build_scripts/build_usd.py

```diff
@@ -226,24 +226,36 @@ def _GetPythonLibraryFilename(context):
         pythonLibPath = os.path.join(pythonBaseDir, "lib",
                                      _GetPythonLibraryFilename(context))
     else:
-        pythonIncludeDir = sysconfig.get_config_var("INCLUDEPY")
-        if Windows():
+        pythonIncludeDir = sysconfig.get_path("include")
+        if not pythonIncludeDir or not os.path.isdir(pythonIncludeDir):
+            # as a backup, and for legacy reasons - not preferred because
+            # it may be baked at build time
+            pythonIncludeDir = sysconfig.get_config_var("INCLUDEPY")
+
+        # if in a venv, installed_base will be the "original" python,
+        # which is where the libs are ("base" will be the venv dir)
+        pythonBaseDir = sysconfig.get_config_var("installed_base")
+        if not pythonBaseDir or not os.path.isdir(pythonBaseDir):
+            # for python-2.7
             pythonBaseDir = sysconfig.get_config_var("base")
-            pythonLibPath = os.path.join(pythonBaseDir, "libs",
-                                         _GetPythonLibraryFilename(context))
+
+        if Windows():
+            pythonLibDir = os.path.join(pythonBaseDir, "libs")
         elif Linux():
-            pythonLibDir = sysconfig.get_config_var("LIBDIR")
+            pythonLibDir = os.path.join(pythonBaseDir, "lib")
+            if not os.path.isdir(pythonLibDir):
+                # as a backup, and for legacy reasons - not preferred because
+                # it may be baked at build time
+                pythonLibDir = sysconfig.get_config_var("LIBDIR")
```

Comment:
This doesn't give me the right results on my RHEL7 workstation running Python 3.6.8. On this system, the Python library is located at `/usr/lib64/libpython3.6m.so`. That's what I get with the old code:

```
sysconfig.get_config_var("LIBDIR") => '/usr/lib64'
pythonLibDir = '/usr/lib64'
pythonLibPath = '/usr/lib64/libpython3.6m.so'
```
but with this change I'd get a different path that doesn't exist:

```
sysconfig.get_config_var("installed_base") => '/usr'
sysconfig.get_config_var("base") => '/usr'
pythonLibDir = '/usr/lib'
pythonLibPath = '/usr/lib/libpython3.6m.so'
```

I wonder if, instead of checking if the *directory* generated via the new method exists, we should check whether the *library* exists instead. If it doesn't, then we fall back to the old method.

Reply:
@sunyab - Sorry for the long response time.  Tried doing as you suggested, and I now check if the library exists, and go onto another method if it doesn't.  Also made it try the old way first, for better backward compatility.